File: AnotherBlogSite.Presentation/ClientApp/models/BlogPost.ts
import Guid from "./Guid";
import User from "./User";
import Comment from "./Comment";

export default interface BlogPost {
    id: Guid;
    title: string;
    content: string;
    createdDate: Date;
    authorId: Guid;
    author: User;
    comments: Comment[] | null;
}

File: AnotherBlogSite.Application/Entities/BlogPost.cs
namespace AnotherBlogSite.Application.Entities;

public sealed class BlogPost
{
    public Guid Id { get; set; }
    public string Title { get; set; }
    public string Content { get; set; }
    public DateTimeOffset CreatedDate { get; set; }

    public Guid AuthorId { get; set; }
    public User Author { get; set; }

    public ICollection<Comment> Comments { get; set; }
}
